pkg/gpu/testutil/mocks.go

```diff
@@ -291,6 +308,54 @@ func WithSymbolsMock(availableSymbols map[string]struct{}) NvmlMockOption {
 	}
 }
 
+// WithMockAllFunctions returns an option that creates basic functions for all nvmlmock.Device.*Func attributes
+// that return nil/zero values. This is useful for ensuring all functions are mocked even if not explicitly set.
+// This is not the default behavior of the mock, as we want explicit errors if we use a function that is not mocked
+// so that we implement the mocked method explicitly, controlling the inputs and outputs. However, in some cases
+// (e.g., testing the collectors) we want to ensure that all functions are mocked without caring too much about the inputs and outputs.
```

Comment:
why in case of testing the collectors we want to have all functions mocked? I would expect a strict mocking as well, to ensure the reliability of the test